### ISSUE TITLE
Getting function argument name

### ISSUE THREAD

**Issue by ales-tsurko** [koto-lang/koto#42]
Is there a way to get a function's argument name? 

In my case I need it to capture arguments from callback, to expose them later as parameters. There would be API, for example:

```coffee
# there is a function to define a node, the arguments are:
# - name of the node
# - output types (audio/control rate per output)
# - input types
# - callback, which will generate a syntax tree from which Csound code will be generated
node_def "synth 1", "aa", "kkk", |freq, amp, mul| 
  l = oscili.a freq, add, mul * 0.5
  r = oscili.a freq + 10, add, mul * 0.5
  out l, r

# node definitions are added into a global dictionary, then I can init one like this
# (we specify types as nodes are polymorphic (= there can be the same type of nodes, 
# but with different types of inputs/outputs))
n = node "synth 1", "aa", "kkk", 440, 0, 0.5

# then I'd like to be able to set parametrs by name, for this I need to get the names
# of the arguments from the callback
n.set "freq", 220
```

I also need the arguments names to generate readable variables in the produced Csound code, but I can go with some random names for that.

**Comment by irh** [comment]
There's currently no way to query a function's argument names. I have a half-baked idea of adding support for debuggers at some point which would need value names to be queryable, but this would be a long way off, and I hadn't thought about making this info available internally in a script.

I'm not sure that I fully understand your example, but I guess if it was me I'd consider declaring node inputs more explicitly, like `(k("freq"), k("amp"), k("mul"))`, or `k("freq", "amp", "mul")`. This would introduce some redundancy but at least it would be more straight forward?

**Comment by ales-tsurko** [comment]
> I'm not sure that I fully understand your example, but I guess if it was me I'd consider declaring node inputs more explicitly, like (k("freq"), k("amp"), k("mul")), or k("freq", "amp", "mul"). This would introduce some redundancy but at least it would be more straight forward?

Yeah, there are options. I just tried to start from the most straightforward 😄 

And there isn't even some ugly way, like somehow getting the callback as a string then parsing it and getting the argument names from the AST?

**Comment by ales-tsurko** [comment]
Nevermind. It turned out that using a map with something like you suggested is actually even more straightforward, then the way I was going to do this. Thanks!

**Comment by irh** [comment]
> And there isn't even some ugly way, like somehow getting the callback as a string then parsing it and getting the argument names from the AST?

Not yet no, but I'd like to see a 'parse, compile and run koto scripts' module for koto at some point.
